shanoir-ng-datasets/src/main/java/org/shanoir/ng/bids/service/BIDSServiceImpl.java

```diff
@@ -392,9 +426,14 @@ private File createExaminationFolder(final Examination examination, final File s
 	 * @return A list of newly created specific BIDS files associated to the dataset in entry
 	 * @throws IOException when we fail to create a file
 	 */
-	private void createDatasetBidsFiles(final Dataset dataset, final File workDir, final String studyName, final String subjectName) throws IOException {
+	private void createDatasetBidsFiles(final Dataset dataset, final File workDir, final String studyName, String subjectName) throws IOException {
 		File dataFolder = null;
 
+		String nature = null;
+		subjectName = subjectName.replaceAll(" ", "");
+		subjectName = subjectName.replaceAll("_", "");
+		String datasetFilePrefix = workDir.getName();
+
 		// Create specific files (EEG, MS, MEG, etc..)
 		if (dataset instanceof EegDataset) {
```

Comment:
I think, in order to limit the size & complexity of the `createDatasetBidsFiles` method, we should put this if..else if..else if... part into a bigger `createDataFolder` method.

So that we just have to call `dataFolder = createDataFolder(dataset, workdir);` here.